apps/android/LiveVideoApp/app/src/main/java/com/twilio/livevideo/app/repository/datasource/remote/RemoteStorage.kt

```diff
@@ -1,53 +1,61 @@
 package com.twilio.livevideo.app.repository.datasource.remote
 
-import com.google.gson.Gson
+import com.google.gson.stream.MalformedJsonException
 import com.twilio.livevideo.app.repository.model.BaseResponse
 import com.twilio.livevideo.app.repository.model.CreateStreamResponse
 import com.twilio.livevideo.app.repository.model.DeleteStreamResponse
+import com.twilio.livevideo.app.repository.model.ErrorResponse
 import com.twilio.livevideo.app.repository.model.JoinStreamAsViewerResponse
 import com.twilio.livevideo.app.repository.model.VerifyPasscodeResponse
-import com.twilio.livevideo.app.util.ApiResponseUtil
 import retrofit2.Response
+import retrofit2.Retrofit
 import javax.inject.Inject
 
-class RemoteStorage @Inject constructor(private var liveVideoAPIService: LiveVideoAPIService) {
-
-    private val gson = Gson()
+class RemoteStorage @Inject constructor(private val liveVideoAPIService: LiveVideoAPIService, private val retrofit: Retrofit) {
 
     suspend fun verifyPasscode(passcode: String): VerifyPasscodeResponse =
-        processResponse(liveVideoAPIService.verifyPasscode(passcode))
+        validateResponse(liveVideoAPIService.verifyPasscode(passcode))
 
     suspend fun joinStreamAsViewer(
         userIdentity: String,
         streamName: String
     ): JoinStreamAsViewerResponse =
-        processResponse(liveVideoAPIService.joinStreamAsViewer(userIdentity, streamName))
+        validateResponse(liveVideoAPIService.joinStreamAsViewer(userIdentity, streamName))
 
     suspend fun createStream(
         userIdentity: String,
         streamName: String
     ): CreateStreamResponse =
-        processResponse(liveVideoAPIService.createStream(userIdentity, streamName))
+        validateResponse(liveVideoAPIService.createStream(userIdentity, streamName))
 
     suspend fun deleteStream(
         streamName: String
     ): DeleteStreamResponse =
-        processResponse(liveVideoAPIService.deleteStream(streamName))
+        validateResponse(liveVideoAPIService.deleteStream(streamName))
 
-    private inline fun <reified T : BaseResponse> processResponse(
+    private inline fun <reified T : BaseResponse> validateResponse(
         response: Response<T>,
     ): T {
-        val result: T = response.body() ?: T::class.java.newInstance()
+        val result: T = if (!response.isSuccessful) { // Failure case
+            response.errorBody()?.let { body ->
+                try {
+                    retrofit.responseBodyConverter<T>(T::class.java, arrayOf()).convert(body)
+                } catch (e: MalformedJsonException) {
+                    null
```

Comment:
this error should be trapped some how and at least logged to logcat

Reply:
Done!